util/topgen/secure_prng.py

```diff
@@ -0,0 +1,289 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+"""Generating random variables by using a cryptographically secure PRNG.
+
+A class for generating random numbers and permutations.
+This class is based on a cryptographically secure PRNG CTR_DRBG described in
+NIST SP 800-90A, section 10.2.1
+https://csrc.nist.gov/publications/detail/sp/800-90a/rev-1/final
+This PRNG guarantees forward and backward secrecy (PRNG output cannot be
+computed from any previous or future outpus) as well as enhanced backward
+secrecy (previous PRNG outputs cannot be computed from a compromised
+internal state).
+"""
+
+import logging as log
+import sys
+from Crypto.Cipher import AES
+from math import ceil as _ceil
+from math import log as _log
+
+
+class secure_prng():
```

Comment:
Can we think of a better name for this? Maybe call it `ctr_drbg_no_input_256` or some variant of that?

Reply:
I would keep the name. IMHO it's descriptive enough and the implementation provides many comments to detail what construction exactly is used under the hood.